### PR TITLE
Research GitHub Workflow Tooling

### LINKED ISSUE #9: Research GitHub Workflow Tooling
Research GitHub's workflow tooling, such as:

- Issues
- Project
- PR
- Discussions
- Milestones 
- Labels

### PR DESCRIPTION
**Pull request by FridaPshitik**
Describe and explain the various GitHub workflow tools:

- GitHub's issues and PR features.
- GitHub's projects.
- GitHub's discussions.
- GitHub's milestones and labels.

Closes #9

### CODE REVIEW

**Comment by bdsoha** [comment]
The work you uploaded is very in depth 👍 .

I don't think that this needs to be a *tutorial guide* or *user manual* on how to click-through and use the tool.
It should be more along the lines of an *overview* so that future users will know **when** and **why** the tool is needed.

You can provide 1 or 2 code snippets or screenshots *(if needed)* to present the overall use-case.

---

Unfortunately, there are a few things missing.

**GitHub's projects** refers to the *project management* tool.
It seems like you may have confused this topic with *repositories* which are designed to manage the project's code.

**Comment by FridaPshitik** [comment]
> The work you uploaded is very in depth 👍 .
> 
> I don't think that this needs to be a _tutorial guide_ or _user manual_ on how to click-through and use the tool. It should be more along the lines of an _overview_ so that future users will know **when** and **why** the tool is needed.
> 
> You can provide 1 or 2 code snippets or screenshots _(if needed)_ to present the overall use-case.
> 
> Unfortunately, there are a few things missing.
> 
> **GitHub's projects** refers to the _project management_ tool. It seems like you may have confused this topic with _repositories_ which are designed to manage the project's code.

**we try to understand the confuesed and fix it.**

**Comment by bdsoha** [comment]
If there is something that you don't understand or is unclear, please ask.
It is a shame to work on something in vain.

**Comment by bdsoha** [comment]
> you mean to do the explanation on the project from the managment side or in general ?

In general.
If you have a question, **ask**.
If something isn't clear, **ask**.

There is no reason for someone to work on anything for 2-days to only then find out that something is missing or incorrect.

**Comment by FridaPshitik** [comment]
we try change the names of the file Manually and also in git bash but It doesn't change.

**Comment by FridaPshitik** [comment]
We went through the files again and changed what we think needs to be changed, 
but we still can't change the file names
We'd love to know if there are any other comments

**Comment by bdsoha** [comment]
> we try change the names of the file Manually and also in git bash but It doesn't change.

That is because Window's filesystem is *case-insensitive* and `git` is based on Linux which is *case-sensitive*.
For example, Windows will not allow you to create a file named `a.txt` and `A.txt`, but Linux will.

Therefore, you need to make a 2-step change:
```txt
A.txt ➡ a.txt.tmp
git commit
a.txt.tmp ➡ a.txt
git commit
```

The `tmp` suffix could be anything.

**Comment by FridaPshitik** [comment]
thanks on the explanation.
we changed the names files.
did you went over the files ?

**Comment by bdsoha** [comment]
> thanks on the explanation. we changed the names files. did you went over the files ?

You will know when I review the files by receiving either an *approval* or *request of changes* to this PR.

**Comment by bdsoha** [comment]
You still did not summarize the correct *projects*.

**Comment by FridaPshitik** [comment]
just for update - we are in the middle of code review.

File: github-workflow/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Research GitHub Workflow Tooling
+
+## GitHub's issues and PR features
+
+[issues and PR features](issues-and-PR-features.md)
```
Comment (RivkaBerlinski) [inline github-workflow/README.md:5]: The name of the file path does not match the file name.

File: github-workflow/issues-and-pr-features.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+- resolved
+
+If you use a keyword to reference a pull request comment in another pull request, the pull requests will be linked.
+
+Merging the referencing pull request also closes the referenced pull request.
+
+Only manually linked pull requests can be manually unlinked. To unlink an issue that you linked using a keyword, you must edit the pull request description to remove the keyword.
+
+You can also use closing keywords in a commit message.
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:57]: You write this in line 27

File: github-workflow/README.md (hunk 1 above)
Comment (FridaPshitik) [reply github-workflow/README.md:5]: we finish to review.
now we are going through comments from Rivka and Chana.

**Comment by FridaPshitik** [comment]
we finish to review.
now we are going through  comments from Rivka and Chana.

File: github-workflow/issues-and-pr-features.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+The issue will be closed when you merge the commit into the default branch, but the pull request that contains the commit will not be listed as a linked pull request.
+
+> Manually linking a pull request to an issue using the pull request sidebar
+
+Anyone with write permissions to a repository can manually link a pull request to an issue from the pull request sidebar.
+
+You can manually link up to ten issues to each pull request. The issue and pull request must be in the same repository.
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:65]: what does it means?

File: github-workflow/issues-and-pr-features.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+### Efficient communication
+
+You can @mention collaborators who have access to your repository in an issue to draw their attention to a comment. To link related issues in the same repository, you can type # followed by part of the issue title and then clicking the issue that you want to link. To communicate responsibility, you can assign issues. If you find yourself frequently typing the same comment, you can use saved replies.
+
+> Create a new issue :
+
+ Write title and description
+
+!['sceernshot issue'''](screenshots/issues-and-PR-features.pic/pic3.png)
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:79]: Routing problem

File: github-workflow/issues-and-pr-features.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+The issue will be closed when you merge the commit into the default branch, but the pull request that contains the commit will not be listed as a linked pull request.
+
+> Manually linking a pull request to an issue using the pull request sidebar
+
+Anyone with write permissions to a repository can manually link a pull request to an issue from the pull request sidebar.
+
+You can manually link up to ten issues to each pull request. The issue and pull request must be in the same repository.
+
+## issue
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:67]: A key sentence that explains in general what it is is missing ``issue``

File: github-workflow/issues-and-pr-features.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+Other contributors can review your proposed changes, add review comments, contribute to the pull request discussion, and even add commits to the pull request.
+
+By default, in public repositories, any user can submit reviews that approve or request changes to a pull request.
+
+Organization owners and repository admins can limit who is able to give approving pull request reviews or request changes.
+
+You can see information about the branch's current deployment status and past deployment activity on the "Conversation" tab.
+
+After you're happy with the proposed changes, you can merge the pull request. If you're working in a shared repository model, you create a pull request and you, or someone else,
+
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:104]: Unnecessary space line

File: github-workflow/issues-and-pr-features.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+If status checks are required for a repository, the required status checks must pass before you can merge your branch into the protected branch.
+
+You can link a pull request to an issue to show that a fix is in progress and to automatically close the issue when someone merges the pull request.
+
+> Creating a pull request
+
+ In the "Branch" menu, choose the branch that contains your commits.
+
+!['screenshot pull request'](screenshots/issues-and-PR-features.pic/pic5.png)
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:115]: Routing problem

File: github-workflow/issues-and-pr-features.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+ Above the list of files, click  Pull request
+
+!['screenshot pull request'](screenshots/issues-and-PR-features.pic/pic6.png)
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:119]: Routing problem

File: github-workflow/issues-and-pr-features.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+You can link a pull request or branch to an issue to show that a fix is in progress and to automatically close the issue when the pull request or branch is merged.
+
+You can link an issue to a pull request manually or using a supported keyword in the pull request description.
+
+When you link a pull request to the issue the pull request addresses, collaborators can see that someone is working on the issue.
+
+When you merge a linked pull request into the default branch of a repository, its linked issue is automatically closed.
+
+> Linking a pull request to an issue using a keyword
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:25]: Mark as a subheading

File: github-workflow/issues-and-pr-features.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+
+Merging the referencing pull request also closes the referenced pull request.
+
+Only manually linked pull requests can be manually unlinked. To unlink an issue that you linked using a keyword, you must edit the pull request description to remove the keyword.
+
+You can also use closing keywords in a commit message.
+
+The issue will be closed when you merge the commit into the default branch, but the pull request that contains the commit will not be listed as a linked pull request.
+
+> Manually linking a pull request to an issue using the pull request sidebar
```
Comment (RivkaBerlinski) [inline github-workflow/issues-and-pr-features.md:61]: Mark as a subheading

File: github-workflow/projects.md [hunk 11]
```diff
@@ -0,0 +1,183 @@
+# About Projects
```
Comment (RivkaBerlinski) [inline github-workflow/projects.md:1]: Unnecessary space lines inside paragraphs

File: github-workflow/projects.md [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+
+Iteration fields also let you view work that you completed in past iterations,
+
+which can help with velocity planning and reflecting on your team's accomplishments.
+
+Iteration fields also support breaks to show when you and your team are taking time away from their iterations.
+
+## Pros and Cons
+
+> The pros :
```
Comment (RivkaBerlinski) [inline github-workflow/projects.md:139]: Mark the advantages as points in the list

File: github-workflow/projects.md [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+
+It is also beneficial to use if you’re learning Git.
+
+One way to look at it is seeing Git as a preventative maintenance process, where you get into the flow of doing reverts, commits, pulling, pushing, and so forth, as needed.
+
+ It can save you a lot of hassle if a problem occurs, or if you need to revert code of any sort.
+
+  One of the best features is being able to track mistakes, and it’s definitely made easier by using GitHub. Bugs can be opened, and thanks to GitHub, you can add commits to these and associate it with the problem.With GitHub pages, you can get started quickly without worrying about setting up hosting or DNS.
+
+  > The cons :
```
Comment (RivkaBerlinski) [inline github-workflow/projects.md:159]: Mark the cons as points in the list

File: github-workflow/projects.md [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+It’s ideal for working on projects of any size and it’s a great tool for web work flows.
+
+Firstly, you can use it for publishing your work, as a versioning control system and as a collaborative tool.
+
+With GitHub you can archive your projects and share them with your peers for either inspiration or for them to collaborate on.
+
+The beauty of it is that you can have more than one person collaborate remotely on a project, so you’re all on the same page, so to speak.It’s ideal for creating a backup of your work instead on saving the code on a desktop or on a corporate server.
+
+It is also beneficial to use if you’re learning Git.
+
```
Comment (RivkaBerlinski) [inline github-workflow/projects.md:152]: Isn't this an unnecessary profit line?

File: github-workflow/projects.md [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+  One of the best features is being able to track mistakes, and it’s definitely made easier by using GitHub. Bugs can be opened, and thanks to GitHub, you can add commits to these and associate it with the problem.With GitHub pages, you can get started quickly without worrying about setting up hosting or DNS.
+
+  > The cons :
+
+  If you’re new to GitHub, one of the challenges often talked about is getting to grips with the mental model, which eases with practice and time.
+
+  It may not be the best tool for capturing the creative process or for recording ideas.
+
+   A good tool for this particular function would be LayerVault or something similar to it.
```
Comment (RivkaBerlinski) [inline github-workflow/projects.md:165]: Is this a disadvantage?

File: github-workflow/discussions.md [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+You want to measure interest in new ideas and gauge community opinion using polls.
+
+You might use repository discussions to discuss topics that are specific to the repository. If your project spans multiple repositories, you might use organization discussions to discuss topics that aren't specific to a single repository in your organization.
+
+## Enabling GitHub Discussions on your repository
+
+Repository owners and people with write access can enable GitHub discussions for a community on their public and private repositories. The visibility of a discussion is inherited from the repository the discussion is created in.
+
+!['screenshots of new discussion'](screenshots/discussions.pic/pic3.png)
```
Comment (RivkaBerlinski) [inline github-workflow/discussions.md:23]: Blurry picture

File: github-workflow/milestones-and-labels.md [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+
+You can use milestones to track progress on groups of issues or pull requests in a repository. When you create a milestone, you can associate it with issues and pull requests. To better manage your project, you can view details about your milestone.
+
+From the mile# Using labels and milestones to track work
+
+## Use labels and milestones to categorize and track work on GitHub
+
+### Milestones
+
+You can use milestones to track progress on groups of issues or pull requests in a repository. When you create a milestone, you can associate it with issues and pull requests. To better manage your project, you can view details about your milestone.
```
Comment (chananaimen) [inline github-workflow/milestones-and-labels.md:14]: The code repeats the code in line 6

File: github-workflow/milestones-and-labels.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+Under "Description", type a description to help others understand and use your label.
+
+!['screenshot of milestones'](screenshots/milestones-and-labels.pic/pic10.png)
+
+> Applying a label
+
+- Navigate to the issue, pull request, or discussion.
+- ln the right sidebar, to the right of "Labels", click, then click a label.
+
+!['screenshot of milestones-and-labels'](screenshots/milestones-and-labels.pic/pic12.png)
```
Comment (RivkaBerlinski) [inline github-workflow/milestones-and-labels.md:56]: Add a transition after the image

File: github-workflow/milestones-and-labels.md [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+stone page, you can see :
```
Comment (RivkaBerlinski) [inline github-workflow/milestones-and-labels.md:57]: This paragraph has already been written before

**Pull request merged**